src/row.js

```diff
@@ -75,24 +91,37 @@ export class Row {
     this.isRanked === 'NO';
 
   isRowClean = () =>
+    this.teamPlayerName1 &&
+    this.teamPlayerName2 &&
     this.enemyPlayerClass1 &&
     this.enemyPlayerClass2 &&
     this.enemyFaction &&
-    this.teamPlayerName1 &&
-    this.teamPlayerName2 &&
     ((this.teamColor && this.winnerColor) || !isNaN(this.diffRating)) &&
     this.isValidSeason();
 
-  matchSummary = () => {
-    const outcome = this.won() ? 'Victory' : 'Defeat';
-    const mmr = this.mmr ? ` at ${this.mmr} MMR` : '';
-    const enemies = `${enemy(
-      this.enemyPlayerName1,
-      this.enemyPlayerClass1
-    )} and ${enemy(this.enemyPlayerName2, this.enemyPlayerClass2)}`;
-    return `${outcome} as ${this.teamPlayerName1}/${this.teamPlayerName2} vs ${enemies}${mmr}`;
-    // could also have shown isRanked/diffRating, day (endTime), zoneId...
-  };
+  is2sData = () =>
+    !this.teamPlayerName3 &&
+    !this.teamPlayerClass3 && // TODO: was previously name only, any reason?
```

Comment:
I don't know... for the filter inside the addon in lua, it uses teamClass only.
See https://github.com/denishamann/ArenaStatsTBC/blob/2ae38102ca24f933d674203aa20094511a4aa146/ArenaStats.lua#L241

Reply:
For enemies, name is sometimes missing, but class is always available?
For allies, afaik, both are always available?
When you had the ghost player bug, did you have additional enemy names AND classes?